airflow/contrib/operators/gcp_transfer_operator.py

```diff
@@ -0,0 +1,856 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+from copy import deepcopy
+from datetime import date, time
+
+from airflow import AirflowException
+from airflow.contrib.hooks.gcp_transfer_hook import GCPTransferServiceHook, \
+    GcpTransferJobsStatus
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+try:
+    from airflow.contrib.hooks.aws_hook import AwsHook
+except ImportError:  # pragma: no cover
+    AwsHook = None
+
+
+class TransferJobPreprocessor:
+
+    def __init__(self, body, aws_conn_id='aws_default'):
+        self.body = body
+        self.aws_conn_id = aws_conn_id
+
+    def _inject_aws_credentials(self):
+        if 'transferSpec' not in self.body or \
+           'awsS3DataSource' not in self.body['transferSpec']:
+            return
+
+        aws_hook = AwsHook(self.aws_conn_id)
+        aws_credentials = aws_hook.get_credentials()
+        aws_access_key_id = aws_credentials.access_key
+        aws_secret_access_key = aws_credentials.secret_key
+        self.body['transferSpec']['awsS3DataSource']["awsAccessKey"] = {
+            "accessKeyId": aws_access_key_id,
+            "secretAccessKey": aws_secret_access_key
+        }
+
+    def _reformat_date(self, field_key):
+        if field_key not in self.body['schedule']:
+            return
+        if isinstance(self.body['schedule'][field_key], date):
+            self.body['schedule'][field_key] = self.\
+                _convert_date(self.body['schedule'][field_key])
+
+    def _reformat_time(self, field_key):
+        if field_key not in self.body['schedule']:
+            return
+        if isinstance(self.body['schedule'][field_key], time):
+            self.body['schedule'][field_key] = self.\
+                _convert_time(self.body['schedule'][field_key])
+
+    def _reformat_schedule(self):
+        if 'schedule' not in self.body:
+            return
+        self._reformat_date('scheduleStartDate')
+        self._reformat_date('scheduleEndDate')
+        self._reformat_time('startTimeOfDay')
+
+    def process_body(self):
+        self._inject_aws_credentials()
+        self._reformat_schedule()
+        return self.body
+
+    @staticmethod
+    def _convert_date(field_date):
+        return {
+            'day': field_date.day,
+            'month': field_date.month,
+            'year': field_date.year
+        }
+
+    @staticmethod
+    def _convert_time(time):
```

Comment:
I think it's not good to name variable the same name as import above - it shadows the import and you are not sure what you use inside. Propose to name it time_ or a_time or smth.

Reply:
What do you think about `time_obj`? I would then change the name of the method to `_convert_time_to_dict`.

Reply:
Fine for me.

Reply:
I changed the name and added a description.